pkg/grub/rootfs.cfg

```diff
@@ -280,6 +290,9 @@ set_to_existing_file efi_grub_cfg "/EFI/BOOT/grub-hv.cfg"
 do_if_args source $efi_grub_cfg
 do_if_args source $config_grub_cfg
 
+# setup bootflow per EVE's flavor
+set_${eve_flavor}_boot
```

Comment:
> This looks circular; eve_flavor was set some e.g., to xen from the image

I actually don't think it is -- but I can agree that it may be surprising (but yes -- it works). Now the only reason it is there is to take care of legacy use cases that may still have a sole `set_kvm_boot` in their `grub.cfg` override.

Making `set_xen_flavor` idempotent is actually quite difficult (although not impossible) -- so we need to treat it as a singleton for now -- and the current code in this PR accommodates that.

So for now, I'd suggest adding this simple tweak in and perhaps provide addition comments in the grub.cfg itself (along the lines of what I put in this PR's description). This way we can unblock ourselves and then somebody who has way more time for very detailed testing on intel and ARM can do the right thing and make `set_xen_flavor` idempotent.

How does that sound?

Reply:
Some additional comments would be helpful